mm2src/mm2_bitcoin/spv_validation/src/conf.rs

```diff
@@ -96,14 +96,47 @@ pub struct SPVConf {
 impl SPVConf {
     pub fn validate(&self, coin: &str) -> Result<(), SPVError> {
         if let Some(params) = &self.validation_params {
-            if let Some(DifficultyAlgorithm::BitcoinMainnet) = &params.difficulty_algorithm {
-                validate_btc_spv_header_height(coin, self.block_header.height)?;
-                if let Some(max) = self.max_stored_block_headers {
-                    validate_btc_max_stored_headers_value(max.into())?;
+            if let Some(algo) = &params.difficulty_algorithm {
+                match algo {
+                    DifficultyAlgorithm::BitcoinMainnet => {
+                        validate_btc_spv_header_height(coin, self.block_header.height)?;
+                        if let Some(max) = self.max_stored_block_headers {
+                            validate_btc_max_stored_headers_value(max.into())?;
+                        }
+                    },
+                    DifficultyAlgorithm::BitcoinTestnet => (),
                 }
             }
         }
 
         Ok(())
     }
+
+    /// Validate Starting block header from `RPC` against [`SPVConf::SPVBlockHeader`]
+    pub fn validate_rpc_starting_block_header(&self, height: u64, rpc_header: &BlockHeader) -> Result<(), SPVError> {
+        let rpc_header = SPVBlockHeader::from_height_and_block_header(height, rpc_header);
+        let spv_header = &self.block_header;
+
+        let rpc_header_bits: u32 = rpc_header.bits.into();
+        let spv_header_bits: u32 = spv_header.bits.clone().into();
+        if rpc_header_bits != spv_header_bits {
+            return Err(SPVError::SPVBlockHeaderError(
+                "Block header not acceptable - bad header bits".to_string(),
```

Comment:
For this and subsequent errors, please provide more details.

Reply:
This is not fixed yet

Reply:
fixed